src/test/java/org/opensearch/searchrelevance/action/scheduledJob/ScheduledJobsIT.java

```diff
@@ -72,20 +75,37 @@ public void testMainActions_whenCreateReadDeleteScheduledExperiment_thenSuccessf
             null,
             ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
         );
+
         Map<String, Object> getScheduledExperimentResultJson = entityAsMap(getScheduledExperimentResponse);
         assertNotNull(getScheduledExperimentResultJson);
         assertEquals(scheduledExperimentId, getScheduledExperimentResultJson.get("_id").toString());
-        Map<String, Object> source = (Map<String, Object>) getScheduledExperimentResultJson.get("_source");
-        assertNotNull(source);
-        assertNotNull(source.get("id"));
-        assertNotNull(source.get("schedule"));
-        assertEquals(experimentId, source.get("id"));
+        Map<String, Object> scheduledExperimentSource = (Map<String, Object>) getScheduledExperimentResultJson.get("_source");
+        assertNotNull(scheduledExperimentSource);
+        assertNotNull(scheduledExperimentSource.get("id"));
+        assertNotNull(scheduledExperimentSource.get("schedule"));
+        assertEquals(experimentId, scheduledExperimentSource.get("id"));
 
         // Here we have to wait until at last one
         Thread.sleep(CRON_JOB_COMPLETION_MS);
```

Comment:
just waiting for a timeout is a bit unreliable. Can we do combined approach - wait for a shorter period (like 30-45 seconds) and then pool after every 5 seconds. This should work faster in typical CI setup. As this is only tests, I think we can leave PR as is and add this combined polling approach to one of future PRs